Lib/test/test_xmlrpc.py

```diff
@@ -1171,6 +1171,56 @@ def test_gzip_decode_limit(self):
         xmlrpclib.gzip_decode(encoded, max_decode=-1)
 
 
+class HeadersServerTestCase(BaseServerTestCase):
+    class RequestHandler(xmlrpc.server.SimpleXMLRPCRequestHandler):
+        test_header = None
+        test_second_header = None
+
+        def do_POST(self):
+            self.__class__.test_header = self.headers.get('x-test')
+            self.__class__.test_second_header = self.headers.get(
```

Comment:
Can you clarify why you need to stash these specifically-named headers on the class like this? Is it because these objects are ephemeral and you don't ever have direct access to them?

In any case, I think it would be preferable to just store the entire list of headers and check that it's what you expect, e.g.:

```python
def do_POST(self):
    self.__class__.headers_posted = copy.deepcopy(self.headers)
    super().do_POST()
```

Then the tests would look like this:

```python
p = xmlrpclib.ServerProxy(URL, headers=[('X-Test', 'foo')])

self.assertEqual(p.pow(6, 8), 6**8)
self.assertEqual(self.RequestHandler.headers_posted, {'X-Test': 'foo'})
```

I think this would be a stronger test, because it not only tests that the header was received, but also that no additional, anomalous headers were added, and it will prevent anyone from accidentally modifying the tests so that the test headers values aren't being cleared between test runs.

Reply:
Well headers are HTTPMessage so they can not be compared with dict. But I can test the keys of the headers.